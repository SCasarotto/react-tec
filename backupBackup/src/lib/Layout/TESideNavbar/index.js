import TESideNavbar from './TESideNavbar'

export default TESideNavbar
