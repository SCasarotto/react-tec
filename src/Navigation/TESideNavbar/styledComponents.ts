import styled from 'styled-components'

export const Container = styled.div<{ sidebarWidth: number }>`
	position: fixed;
	top: 0px;
	bottom: 0px;
	left: 0px;
	max-height: 100vh;
	height: 100%;
	width: ${(props) => props.sidebarWidth}px;
	display: flex;
	flex-direction: column;
	z-index: 20;
	background-color: ${(props) => props.theme.white};
	overflow-y: auto;
	/*TODO: Investigate if this should be repleaced by one of the theme shadows*/
	box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 15px 0px, rgba(0, 0, 0, 0.05) 1px 0px 5px 0px;
`
export const Logo = styled.img`
	object-fit: contain;
	width: 100%;
	padding: 30px 60px;
`
export const NavContainer = styled.div`
	height: 100%;
`
export const MainUl = styled.ul`
	margin: 0px;
	padding: 0px;
`
