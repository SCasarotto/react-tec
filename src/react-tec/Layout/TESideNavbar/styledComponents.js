import styled from 'styled-components'

export const Container = styled.div`
	${(props) => {
		const { theme, sidebarWidth } = props
		return `
			position: fixed;
			top: 0px;
			bottom: 0px;
			left: 0px;
			max-height: 100vh;
			height: 100%;
			width: ${sidebarWidth}px;
			display: flex;
			flex-direction: column;
			z-index: 20;
			background-color: ${theme.white};
			overflow-y: auto;
			box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 15px 0px, rgba(0, 0, 0, 0.05) 1px 0px 5px 0px;
		`
	}}
`
export const Logo = styled.img`
	${(props) => {
		return `
			object-fit: contain;
			width: 100%;
			padding: 30px 60px;
		`
	}}
`
export const NavContainer = styled.div`
	${(props) => {
		return `
			height: 100%;
		`
	}}
`
export const MainUl = styled.ul`
	${(props) => {
		return `
			margin: 0px;
			padding: 0px;
		`
	}}
`
