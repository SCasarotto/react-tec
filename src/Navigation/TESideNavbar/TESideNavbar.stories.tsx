import React from 'react'
import { Story, Meta } from '@storybook/react'

import { TESideNavbar, TESideNavbarProps } from './TESideNavbar'

export default {
	title: 'Navigation/TESideNavbar',
	component: TESideNavbar,
	argTypes: {},
} as Meta

const Template: Story<TESideNavbarProps> = (args) => <TESideNavbar {...args} />

export const Default = Template.bind({})
Default.args = {}
