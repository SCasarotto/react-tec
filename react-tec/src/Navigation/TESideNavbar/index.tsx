import React from 'react'
import { Link } from 'react-router-dom'

import TESideNavLink, { TESideNavLinkProps } from '../TESideNavLink'

import { Container, Logo, NavContainer, MainUl } from './styledComponents'

export interface TESideNavbarProps {
	sidebarWidth?: number
	Header?: React.ReactNode
	logo?: string
	links?: TESideNavLinkProps[]
	className?: string
}
const TESideNavbar: React.FC<TESideNavbarProps> = (props) => {
	const { sidebarWidth = 200, logo, Header, links, className = '' } = props
	return (
		<Container
			sidebarWidth={sidebarWidth}
			className={`TESideNavbar ${className}`}>
			{Header}
			{logo && !Header && (
				<Link to="/" className="TESideNavbarLogoLink">
					<Logo src={logo} alt="logo" className="TESideNavbarLogo" />
				</Link>
			)}
			<NavContainer className="TESideNavbarNavContainer">
				<MainUl className="TESideNavbarMainUl">
					{links &&
						links.map((link, index) => {
							const {
								title,
								activePath,
								to,
								children,
								exact,
							} = link
							return (
								<TESideNavLink
									key={index}
									to={to}
									activePath={activePath || to}
									title={title}
									exact={exact}
									className="TESideNavbarNavLink">
									{children}
								</TESideNavLink>
							)
						})}
				</MainUl>
			</NavContainer>
		</Container>
	)
}

export default TESideNavbar
