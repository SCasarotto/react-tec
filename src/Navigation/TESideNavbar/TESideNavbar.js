import React from 'react'
import PropTypes from 'prop-types'

import TELink from './../TELink'
import TESideNavLink from './../TESideNavLink'

import { Container, Logo, NavContainer, MainUl } from './styledComponents'

const TESideNavbar = (props) => {
	const { sidebarWidth, location, logo, links, className } = props
	return (
		<Container sidebarWidth={sidebarWidth} className={className}>
			{logo && (
				<TELink to="/">
					<Logo src={logo} alt="brandmark" title="Rittal" />
				</TELink>
			)}
			<NavContainer>
				<MainUl>
					{links &&
						links.map((link, index) => {
							const { title, activePath, to, children, exact } = link
							return (
								<TESideNavLink
									key={index}
									to={to}
									activePath={activePath || to}
									title={title}
									exact={exact}
								>
									{children}
								</TESideNavLink>
							)
						})}
				</MainUl>
			</NavContainer>
		</Container>
	)
}

TESideNavbar.propTypes = {
	sidebarWidth: PropTypes.number,
	logo: PropTypes.string,
	links: PropTypes.array,
}

TESideNavbar.defaultProps = {
	sidebarWidth: 200,
}

export default TESideNavbar
