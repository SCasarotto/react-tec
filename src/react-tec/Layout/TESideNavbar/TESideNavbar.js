import React from 'react'
import { withRouter } from 'react-router'
import { Link } from 'react-router-dom'

import TESideNavLink from './../../Navigation/TESideNavLink'

import { Container, Logo, NavContainer, MainUl } from './styledComponents'

const TESideNavbar = (props) => {
	const { sidebarWidth, location, logo, links } = props
	return (
		<Container sidebarWidth={sidebarWidth}>
			{logo && (
				<Link to="/">
					<Logo src={logo} alt="brandmark" title="Rittal" />
				</Link>
			)}
			<NavContainer>
				<MainUl>
					{links &&
						links.map((link, index) => {
							const { title, activePath, to, children } = link
							return (
								<TESideNavLink
									key={index}
									to={to}
									activePath={activePath || to}
									location={location}
									title={title}
								>
									{children}
								</TESideNavLink>
							)
						})}
				</MainUl>
			</NavContainer>
		</Container>
	)
}

TESideNavbar.defaultProps = {
	sidebarWidth: 200,
}

export default withRouter(TESideNavbar)
