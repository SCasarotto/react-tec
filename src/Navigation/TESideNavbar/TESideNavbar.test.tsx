import React from 'react'
import ReactDOM from 'react-dom'
import { TESideNavbar } from './TESideNavbar'

describe('TESideNavbar', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<TESideNavbar />, div)
		ReactDOM.unmountComponentAtNode(div)
	})
})
