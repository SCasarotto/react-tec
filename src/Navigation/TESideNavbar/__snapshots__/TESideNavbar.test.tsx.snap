// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TESideNavbar matches snapshot 1`] = `
<DocumentFragment>
  .c0 {
  position: fixed;
  top: 0px;
  bottom: 0px;
  left: 0px;
  max-height: 100vh;
  height: 100%;
  width: 200px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  z-index: 20;
  overflow-y: auto;
  box-shadow: rgba(0,0,0,0.1) 0px 0px 15px 0px,rgba(0,0,0,0.05) 1px 0px 5px 0px;
}

.c1 {
  height: 100%;
}

.c2 {
  margin: 0px;
  padding: 0px;
}

<div
    class="c0 TESideNavbar "
  >
    <div
      class="c1 TESideNavbarNavContainer"
    >
      <ul
        class="c2 TESideNavbarMainUl"
      />
    </div>
  </div>
</DocumentFragment>
`;
